modules/sw_services/device_control/host/device_access_spi_rpi.c

```diff
@@ -13,15 +13,29 @@
 #define DBG(x)
 #define PRINT_ERROR(...)   fprintf(stderr, "Error  : " __VA_ARGS__)
 
+// Number of usec to delay between spi transactions
+static long intertransaction_delay;
+
+// Sleep for intertransaction_delay nanoseconds. Yields to the OS so expect minimum delay to be hundreds
+// of microseconds at least.
+static void apply_intertransaction_delay() {
+	if(0 < intertransaction_delay) {
+		struct timespec req = {0, intertransaction_delay};
+		struct timespec rem;
+		while(nanosleep(&req, &rem));
+	}
+}
+
 control_ret_t
-control_init_spi_pi(spi_mode_t spi_mode, bcm2835SPIClockDivider clock_divider)
+control_init_spi_pi(spi_mode_t spi_mode, bcm2835SPIClockDivider clock_divider, long delay)
```

Comment:
It'll be clearer if we specified the units in which we expect user to provide the delay, like delay_us or delay_ns.